nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -817,6 +999,75 @@ impl BackgroundTask for SwitchPortSettingsManager {
                             .unwrap_or(SwitchLinkSpeed::Speed100G)
                             .into(),
                     };
+
+                    for peer in port_config.bgp_peers.iter_mut() {
+                        peer.communities = match self
+                            .datastore
+                            .communities_for_peer(
+                                opctx,
+                                port.port_settings_id.unwrap(),
+                                &peer.port,
+                                IpNetwork::from(IpAddr::from(peer.addr))
+                            ).await {
+                                Ok(cs) => cs.iter().map(|c| c.community.0).collect(),
+                                Err(e) => {
+                                    error!(log,
+                                        "failed to get communities for peer";
+                                        "peer" => ?peer,
+                                        "error" => ?e,
+                                    );
+                                    continue;
```

Comment:
is `continue` right? seems like it should actually error out.

Reply:
This is a general pattern in this part of the code, and I'd like to defer an overall cleanup with @internet-diglett involved.

Reply:
The same applies to the continue / parallelization comments that follow.